### ISSUE TITLE
Fix automated Contributors.md update

### ISSUE THREAD

**Issue by RNKuhns** [sktime/sktime#4103]
Whoops, it currently fails with the following:
```
[main 46fc2ca] [AUTOMATED] update CONTRIBUTORS.md
 Author: achieveordie <achieveordie@users.noreply.github.com>
 1 file changed, 53 insertions(+), 33 deletions(-)
INPUT_TAGGING_MESSAGE: 
No tagging message supplied. No tag will be added.
INPUT_PUSH_OPTIONS: 
remote: error: GH006: Protected branch update failed for refs/heads/main.        
remote: error: At least 1 approving review is required by reviewers with write access. 17 of 17 required status checks are expected.        
To https://github.com/sktime/sktime
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/sktime/sktime'
```

__Originally posted by @achieveordie in https://github.com/sktime/sktime/pull/3807#issuecomment-1382086216__

**Comment by RNKuhns** [comment]
Am I understanding it right, that this fails b/c main is protected branch?

Could we just have it open a new PR?

These should be pretty easy approvals.

**Comment by fkiraly** [comment]
> Am I understanding it right, that this fails b/c main is protected branch?

I don't "know" know, but given the error message it feels like a pretty safe bet.

**Comment by fkiraly** [comment]
> Could we just have it open a new PR?

Could it instead just push to the branch that triggers it, pre-merge?

**Comment by achieveordie** [comment]
We could follow the route [recommended](https://github.com/stefanzweifel/git-auto-commit-action#push-to-protected-branches) by the author of this GH Action.

More specifically (emphasis mine):
> **You need a Personal Access Token and** you either have to allow force pushes or **the Personal Access Token needs to belong to an Administrator**.
> ...
> First, you have to create a new [Personal Access Token (PAT)](https://github.com/settings/tokens/new), store the token as a secret in your repository and pass the new token to the [actions/checkout](https://github.com/actions/checkout#usage) Action step.
> ...
> Note: If you're working in an organisation, and you don't want to create the PAT from your personal account, we recommend using a bot-account for such tokens.

**Comment by achieveordie** [comment]
> Could it instead just push to the branch that triggers it, pre-merge?

This is what I originally intended. [This](https://github.com/stefanzweifel/git-auto-commit-action/issues/211) issue might be helpful in that regard. WDYT? @RNKuhns @fkiraly

**Comment by fkiraly** [comment]
I don't think it's a good idea to use a PAT with admin credentials!

Because that exposes the PAT as a secret to alternative use by users with write access, and by using the admin PAT maliciously, users with lower security clearance could find a way gain admin credentials - PAT are not "purpose-bound".

I would prefer the solution where we push to the triggering branch, or at third branch that does not have branch protection.

**Comment by achieveordie** [comment]
I agree, especially since there's a safer alternative. I'll modify it to add pre-merge, though my changes would have to be merged for us to test it (see the link in my comment above to know why).

**Comment by fkiraly** [comment]
well, we'll have some weeks time to unmerge if it breaks anything, until 0.16.0

**Comment by achieveordie** [comment]
Turns out that things are (somewhat) working. See #4122 for reference. 

In the current workflow, the `on-push` trigger is enabling `update_contributors.yml` workflow to run on the [fork](https://github.com/CodeSmithDSMLProjects/sktime/actions/runs/3944515565/jobs/6750507924) (one can see it is running only on the fork by seeing the repo the workflow checks out, not to mention the job is visible only in the fork)

I'm saying "somewhat" working since it doesn't include the latest changes but the ones made before. I could not find the three contributors in the built `CONTRIBUTORS.md` even though they have added themselves in `.all-contributorsrc` but I can see that my contributions have been reduced from changes to `.all-contributorsrc` in #3807.

**Comment by achieveordie** [comment]
@fkiraly @RNKuhns Apologies for letting this issue become stale, I've been incredibly busy with work lately. Do you think we can finish this up? As per my above comment, the current workflow is working, we just need to discuss if it working well.

If you need a refresher about the current status please let me know.

**Comment by fkiraly** [comment]
that would be appreciated - what would the required actions be?
Is there something that needs approval or merge? The issue #4123 is still a draft (and says "do not merge"...)

**Comment by achieveordie** [comment]
Here's the current status:

1. The original PR #3807 was intended to work during the merge commit. The [error](https://github.com/sktime/sktime/issues/4103#issue-1532665455) shown in the top comment shows a problem with permissions.
2. To counter this, I proposed an alternative in #4123 where we check out the PR repo and run the workflow pre-merge. The author of this action recommended this method but it [introduces](https://github.com/stefanzweifel/git-auto-commit-action/issues/211#issuecomment-1386592664) a security vulnerability (hence a 'DO NOT MERGE' warning).
3. While I was trying to find a third viable alternative, I [found](https://github.com/sktime/sktime/issues/4103#issuecomment-1386533283) that the current workflow is actually working, just not in the way we were expecting.
   a. If you open the linked PR (#4122), you'll see commits made by the bot. The workflow is running on the fork (and hence we can't see any workflow status here, but it is visible on their fork).
   b. There's a catch though, it doesn't include the latest addition, but the last-made changes (Example: Let's say that I change my contributions in `.all-contributorsrc`. In the following week, you add yourself in `.all-constributorsrc` in a different PR. It will be in this PR that the workflow will be triggered (on your fork) to update MY contributions in `CONTRIBUTORS.md`; changes corresponding to you will be reflected in the subsequent PR that makes changes to `.all-contributorsrc`).

What we need to discuss is if we want to rectify this catch. I don't know if there's a direct solution to this but we can consider this issue solved if we don't mind this small catch. We've already spent a lot of time on this addition, the ROI seems pretty low.

If you need further clarification, let me know.

**Comment by fkiraly** [comment]
I don't mind the catch, this is basically just a delay in the workflow run, right?

Does it mean the workflow that is currently on main is working, and we can close this issue as done?

**Comment by achieveordie** [comment]
I'd like to say that this issue can be closed but let me make a PR to test it and we can close it from there.
